Library/OcCpuLib/OcCpuLib.c

```diff
@@ -1207,10 +1207,12 @@ ScanAmdProcessor (
 
     switch (Cpu->ExtFamily) {
       case AMD_CPU_EXT_FAMILY_17H:
-        CofVid           = AsmReadMsr64 (K10_PSTATE_STATUS);
-        CoreFrequencyID  = BitFieldRead64 (CofVid, 0, 7);
-        CoreDivisorID    = BitFieldRead64 (CofVid, 8, 13);
-        Cpu->MaxBusRatio = (UINT8) (CoreFrequencyID / CoreDivisorID * 2);
+        if (!Cpu->CPUFrequencyFromVMWare){
+          CofVid           = AsmReadMsr64 (K10_PSTATE_STATUS);
+          CoreFrequencyID  = BitFieldRead64 (CofVid, 0, 7);
+          CoreDivisorID    = BitFieldRead64 (CofVid, 8, 13);
+          Cpu->MaxBusRatio = (UINT8) (CoreFrequencyID / CoreDivisorID * 2);
```

Comment:
This is unsafe divide, should we change it in this PR?

Reply:
Yes, please do it.